prow/plugins/updateconfig/updateconfig.go

```diff
@@ -105,20 +105,20 @@ func Update(fg FileGetter, kc corev1.ConfigMapInterface, name, namespace string,
 		cm.Data = map[string]string{}
 	}
 
-	for key, filename := range updates {
-		if filename == "" {
-			logger.WithField("key", key).Debug("Deleting key.")
-			delete(cm.Data, key)
+	for _, upd := range updates {
```

Comment:
Previously we would de-dupe keys by virtue of having a map -- do we keep that behavior with the slice?

Reply:
Would it be worse to have

```go
type ConfigMapUpdate struct {
    Filename string
    GZIP bool
}

var toUpdate map[string]ConfigMapUpdate
```

Reply:
The existing behavior is maintained by https://github.com/kubernetes/test-infra/pull/11765/files#diff-ac6db59f9774f9c51329fc7a401ae966R132

We could use another map here, but this avoids a double nested map and slices don't have to be initialized before adding elements like maps do.
/shrug